src/main/java/CurrencyController.java

```diff
@@ -0,0 +1,20 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.math.BigDecimal;
+
+public class CurrencyController {
+    HttpUrlConnection connection = new HttpUrlConnection();
+
+    public CurrencyResponse getAllcurrency() throws Exception {
```

Comment:
usage of client should return you object, most of this nasty work will be done behind the hood :D